### PR TITLE
Adding twine and updating s3 sync

### PR DESCRIPTION
**Pull request by nhibberd**
First of a few small changes (updating multipart upload / download / copy) with latest twine code

### CODE REVIEW

**Comment by nhibberd** [comment]
@charleso @markhibberd @jystic

File: mismi-s3/test/Test/Reliability/Mismi/S3/Commands.hs [hunk 1]
```diff
@@ -28,9 +28,11 @@ import           Test.Reliability.Reliability
 import           Test.QuickCheck
 import           Test.QuickCheck.Instances ()
 
+import           X.Control.Monad.Trans.Either (runEitherT)
+
 prop_sync = forAll (elements muppets) $ \m -> testAWS' $ \a b i -> do
   createFiles a m i
-  syncWithMode OverwriteSync a b 10
+  _ <- runEitherT $ syncWithMode OverwriteSync a b 10
```
Comment (charleso) [inline mismi-s3/test/Test/Reliability/Mismi/S3/Commands.hs]: Ignoring errors? Should we at least fail on left?

File: mismi-s3/src/Mismi/S3/Internal.hs [hunk 2]
```diff
@@ -77,6 +82,10 @@ sinkChanWithDelay :: MonadIO m => Int -> Source m a -> Chan a -> m Int
 sinkChanWithDelay delay source c =
   source $$ DC.foldM (\i v -> liftIO $ threadDelay delay >> writeChan c v >> pure (i + 1)) 0
 
+sinkQueue :: Env -> Source AWS a -> Queue a -> IO ()
+sinkQueue e source q =
+  rawRunAWS e (source $$ DC.mapM_ (liftIO . writeQueue q))
```
Comment (charleso) [inline mismi-s3/src/Mismi/S3/Internal.hs]: Just `rawRunAWS` here? (I'm assuming it's correct, but just checking)

**Comment by charleso** [comment]
Surface :+1: Assuming :green_apple:

File: mismi-s3/src/Mismi/S3/Internal.hs (hunk 2 above)
Comment (nhibberd) [inline mismi-s3/src/Mismi/S3/Internal.hs]: Yeah no way to avoid it, needs to be in IO

**Comment by nhibberd** [comment]
https://github.com/ambiata/twine/issues/13

**Comment by markhibberd** [comment]
:+1:

**Comment by nhibberd** [comment]
Performance testing puts this inline with current implementation

**Comment by nhibberd** [comment]
https://github.com/ambiata/twine/issues/15

File: mismi-s3/src/Mismi/S3/Internal.hs (hunk 2 above)
Comment (nhibberd) [inline mismi-s3/src/Mismi/S3/Internal.hs]: This is wrong, should not use rawRunAWS here

**Comment by nhibberd** [comment]
On top of #242

**Pull request merged**